### PR TITLE
feat: add failing validator name to context errors

### LINKED ISSUE #1030: How to get failed error type
Hi i'm trying to get the error type failed to validate:

This is what o have when validation fail but i want to know what kind of fail was. like: required or maxlength etc.
{
    value: undefined,
    msg: 'Invalid value',
    param: 'lname',
    location: 'body'
  }

I need something like this:

{
    value: undefined,
    msg: 'Invalid value',
    param: 'lname',
    location: 'body',
    king: 'required'
  }

### PR DESCRIPTION
**Pull request by fedeci**
## Description
Closes #1030 

## To-do list

- [x] I have added tests for what I changed.

- [x] This pull request is ready to merge.

### CODE REVIEW

File: src/context-items/custom-validation.ts [hunk 1]
```diff
@@ -17,14 +17,19 @@ export class CustomValidation implements ContextItem {
       // A promise that was resolved only adds an error if negated.
       // Otherwise it always suceeds
       if ((!isPromise && failed) || (isPromise && this.negated)) {
-        context.addError(this.message, value, meta);
+        context.addError(this.message, value, meta, 'customValidator');
```
Comment (gustavohenke) [inline src/context-items/custom-validation.ts:20]: How satisfied are you with `customValidator` here? 🤔

**Review by gustavohenke — COMMENTED** [review]
This will be a breaking change - which to avoid we should implement behind a setting.

Do you think there are any use cases other than making the default error message more useful?

I'm worried of introducing a way for hackers to tell exactly which implementation can be exploited to get through any HTTP call.

Also, extra download weight that may not be necessary to many.

File: src/context-items/custom-validation.ts (hunk 1 above)
Comment (fedeci) [reply src/context-items/custom-validation.ts:20]: No idea of how to improve🤔, maybe `.name` fallbacking to `'customValidator'`?

**Comment by fedeci** [comment]
Why a breaking change? We are not radically changing an existing feature, but adding a new one.

Probably making it opt-in as part of `validationResult` custom formatters is better.

**Pull request closed without merging**